avocado/plugins/envvars.py

```diff
@@ -0,0 +1,69 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2016
+# Author: Amador Pahim <apahim@redhat.com>
+
+import argparse
+import os
+from .base import CLI
+from avocado.core.settings import settings
+
+
+class EnvVars(CLI):
+
+    """
+    Set environment variables
+    """
+
+    name = 'envvars'
+    description = "Set variables in tests processes environemnt"
+
+    def configure(self, parser):
+        run_subcommand_parser = parser.subcommands.choices.get('run', None)
+        if run_subcommand_parser is None:
+            return
+
+        msg = 'environemnt variables'
+        replay_parser = run_subcommand_parser.add_argument_group(msg)
+        replay_parser.add_argument('--test-env', dest='test_env',
+                                   default=None,
+                                   type=self._parse_test_env_vars,
+                                   help='Set environemnt variables for tests')
+
+    def _parse_test_env_vars(self, string):
+        # Convert this:
+        #  'key=value,key2=value2,key3=value3'
+        # into this:
+        #  {'key': 'value', 'key2': 'value2', 'key3': 'value3'}
+        try:
+            return {item.split('=')[0]: item.split('=')[1]
+                    for item in string.split(',')}
```

Comment:
I have to say I'm glad we only keep 2.6 compatibility, it was painful in autotest with 2.4 one...